### ISSUE TITLE
Tree View API remove elements from trees via getTreeItem

### ISSUE THREAD

**Issue by otressler** [microsoft/vscode#94573]
Hey there,

TLDR version in the beginning: Allow removing a Tree Item from a Tree by returning null when getTreeItem is called.

I noticed that the Tree Data Provider API is a bit clunky when I want to remove an item from a tree. My scenario is the following: I have an inline refresh button on a tree item, that requests data about the clicked item and triggers the onDidChangeTreeData event with the clicked item as the target afterwards. It could happen, that the item the server is returning does not actually exist anymore. In that case I'd prefer to return null (or undefined) on the getTreeItem method to remove the item I have clicked on. For now I'm using onDidChangeTreeData on the parent item of the clicked element, such that getChildren just wouldn't return the missing item, but that causes a lot of unwanted traffic.

Thanks in advance

**Comment by alexr00** [comment]
Hmm. I see what you're saying about the extra traffic.
